src/software/multithreading/observer_test.cpp

```diff
@@ -55,7 +55,18 @@ TEST(Observer, getDataReceivedPerSecond_time_buffer_filled)
         test_observer.receiveValue(i);
         std::this_thread::sleep_for(std::chrono::milliseconds(10));
     }
-    EXPECT_NEAR(test_observer.getDataReceivedPerSecond(), 5 / (4 * 0.01), 10);
+    EXPECT_NEAR(test_observer.getDataReceivedPerSecond(), 5 / (4 * 0.01), 20);
+}
+
+TEST(Observer, getDataReceivedPerSecond_time_buffer_filled_twice_over)
+{
+    TestObserver test_observer;
+    for (unsigned int i = 0; i < TestObserver::TIME_BUFFER_SIZE * 2; i++)
+    {
+        test_observer.receiveValue(i);
+        std::this_thread::sleep_for(std::chrono::milliseconds(5));
+    }
+    EXPECT_NEAR(test_observer.getDataReceivedPerSecond(), 5 / (4 * 0.005), 20);
```

Comment:
Could you explain why the actual value is 5 / (4 * 0.005)? Shouldn't it be (TIME_BUFFER_SIZE * 2) / (4 * 0.005) ?

Reply:
This is because the ```TIME_BUFFER_SIZE``` (5) is the maximum amount of elements in the ```time_buffer``` used by the ```observer``` class so it will only look at the last 5 elements given to determine ```DataReceivedPerSecond```

Reply:
hm digging into this some more, it doesn't make much sense for the tests to depend on buffer size. Let me know if the new changes make any sense